modules/economy-system/commands/economy-system.js

```diff
@@ -0,0 +1,378 @@
+const {balance, createleaderboard} = require('../economy-system');
+
+module.exports.beforeSubcommand = async function (interaction) {
+    interaction.str = interaction.client.configurations['economy-system']['strings'];
+    interaction.config = interaction.client.configurations['economy-system']['config'];
+};
+
+module.exports.subcommands = {
+    'work': async function (interaction) {
+        const model = interaction.client.models['economy-system']['cooldown'];
+        const cooldownModel = await model.findOne({
+            where: {
+                userId: interaction.user.id,
+                command: 'work'
+            }
+        });
+        if (cooldownModel) {
+            return interaction.reply({
+                content: interaction.str['cooldown'],
+                ephemeral: true
+            });
+        }
+        const cooldown = await model.create({
+            userId: interaction.user.id,
+            command: 'work'
+        });
```

Comment:
Duplicated code, try creating a new (private) helper-function which accepts the type of the cooldown as parameter and returns a `Promise<Boolean>` which should describe if the command should get executed further